File: app/src/main/java/com/nhatminh/example/architecture/demoarchitecture/model/GithubRepos.java
package com.nhatminh.example.architecture.demoarchitecture.model;

import com.google.gson.annotations.SerializedName;

public class GithubRepos {

    @SerializedName("id")
    private Integer id;

    @SerializedName("name")
    private String name;

    @SerializedName("full_name")
    private String fullName;

    @SerializedName("private")
    private Boolean _private;

    @SerializedName("description")
    private String description;

    @SerializedName("updated_at")
    private String updatedAt;

    @SerializedName("size")
    private Integer size;

    @SerializedName("stargazers_count")
    private Integer stargazersCount;

    @SerializedName("language")
    private String language;

    @SerializedName("has_wiki")
    private Boolean hasWiki;

    @SerializedName("archived")
    private Boolean archived;

    @SerializedName("score")
    private Double score;


    public GithubRepos() {
        id = 0;
        name = "";
        fullName="";
        _private = true;
        description="";
        updatedAt = "";
        size=0;
        stargazersCount=0;
        language="";
        hasWiki=true;
        archived=true;
        score=0d;
    }

    public Integer getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getFullName() {
        return fullName;
    }

    public Boolean get_private() {
        return _private;
    }

    public String getDescription() {
        return description;
    }

    public String getUpdatedAt() {
        return updatedAt;
    }

    public Integer getSize() {
        return size;
    }

    public Integer getStargazersCount() {
        return stargazersCount;
    }

    public String getLanguage() {
        return language;
    }

    public Boolean getHasWiki() {
        return hasWiki;
    }

    public Boolean getArchived() {
        return archived;
    }

    public Double getScore() {
        return score;
    }
}

File: app/src/main/java/com/nhatminh/example/architecture/demoarchitecture/model/GithubRepos.kt
package com.nhatminh.example.architecture.demoarchitecture.model

import com.google.gson.annotations.SerializedName

data class GithubRepos (

        @SerializedName("id")
        var id : Int = -1,

        @SerializedName("full_name")
        var fullName : String = "",

        @SerializedName("private")
        var private : Boolean = false,

        @SerializedName("description")
        var description : String = "",

        @SerializedName("updated_at")
        var updatedAt : String = "",

        @SerializedName("size")
        var size : Int = -1,

        @SerializedName("stargazers_count")
        var stargazersCount : Int = -1,

        @SerializedName("language")
        var language : String = "",

        @SerializedName("has_wiki")
        var hasWiki : Boolean = false,

        @SerializedName("archived")
        var archived : Boolean = false,

        @SerializedName("score")
        var score : Double = 0.0
)